### PR TITLE
Add GenericIrController device

### LINKED ISSUE #407: [FEATURE]-Generic IR Device
**Is your feature request related to a problem? Please describe.**
It is difficult to add an IR device to Essentials. The only way at the moment is to create a new plugin for the IR device, which is a bit of a lift for an IR-controlled device.

**Describe the solution you'd like**
I plan on creating a generic IR device that will take the configured IR file, look at it's contents, then generate a join map on the fly to map all functions in the IR file to joins on a bridge.

### PR DESCRIPTION
**Pull request by andrew-welker**
close #407 
This PR adds the GenericIrController device to allow for loading and bridging ANY IR file to SIMPL. The commands for the IR file are also available as a string array for other Essentials devices to use.

### CODE REVIEW

**Comment by andrew-welker** [comment]
Also closes #406 
Adds PepperDash_Essentials_Core as a project reference for Essentials_Devices_Common & PepperDashEssentials.

File: essentials-framework/Essentials Core/PepperDashEssentialsBase/Devices/GenericIRController.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        public GenericIrController(string key, string name, IrOutputPortController irPort) : base(key, name)
+        {
+            _port = irPort;
+
+            if (_port == null)
+            {
+                Debug.Console(0, this, Debug.ErrorLogLevel.Error, "IR Port is null, device will not function");
+                return;
+            }
+            DeviceManager.AddDevice(_port);
```
Comment (ndorin) [inline essentials-framework/Essentials Core/PepperDashEssentialsBase/Devices/GenericIRController.cs:33]: Need to double check that the IRPortController doesn't already add itself to the DeviceManager

**Review by ndorin — COMMENTED** [review]
We should double check if the IRPortController already adds itself to the DeviceManager in the GetIrPortController method. I vaguely recall seeing that it did in recent exploration.

File: essentials-framework/Essentials Core/PepperDashEssentialsBase/Devices/GenericIRController.cs (hunk 1 above)
Comment (andrew-welker) [reply essentials-framework/Essentials Core/PepperDashEssentialsBase/Devices/GenericIRController.cs:33]: Just checked, and it does not. I'll make an issue to change that, as it affects more than just this class.

**Pull request merged**